src/ui/map.js

```diff
@@ -2892,7 +2923,15 @@ class Map extends Camera {
         // A task queue callback may have fired a user event which may have removed the map
         if (this._removed) return;
 
-        this._updateProjection();
+        if (this.getProjection().name === 'globe') {
+            if (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
+                if (this.transform.projection.name === 'globe') {
+                    this._updateProjection();
+                }
+            } else if (this.transform.projection.name === 'mercator') {
+                this._updateProjection();
+            }
+        }
```

Comment:
Any way this logic be contained and simplified within `_updateProjection` to be closer to context?

Reply:
Similar to my comment in `style.js`, this logic determines when a projection update is needed on render (when passing the zoom threshold). It couldn't go in `map._updateProjection()` because the check is irrelevant during projection changes for any other reason.

It could be wrapped in its own function (as it was previously), but this is the only place the function would be called.

Reply:
Have you investigated whether it is possible to split `this._updateProjection` into separate functions accommodating for the case we intend to do from the callee?
- One for the stylesheet/user update
- One function for the transition switch for explicit projection

I'm a bit concerned about the complexity of these checks, because if we expand/inline what we're doing for this specific case, we end up going through a lot of redundancy:
```
if (this.getProjection().name === 'globe') {
  if (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX) {
    if (this.transform.projection.name === 'globe') {
      ... from within _updateProjection()
      const prevProjection = this.getProjection();
      ...
      if (projection && projection.name === "globe") {
        projection = {name: (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX ? 'mercator' : 'globe')};
      }
      ...
```

Reply:
In my first attempt, the projection was always passed into the equivalent of `_updateProjection` and in this case we could instead call a similar function `_updateGlobeTransition` that could skip the redundant checks for globe and zoom.

But by doing so, we would lose the advantages to having all projection changes happen in one function. Most important of which is that this ensures that projection updates return consistent results from application state, regardless of the function that modified the state. This is helpful for avoiding buggy edge cases that can be difficult to think through or test for all possible combinations. For example, prior to these changes, updating the projection to globe by changing the style spec would always render globe instead of Mercator at high zooms.

I've made some changes to `_updateProjection` and replaced some of the if statements with nested ternary operators. Let me know if you think that helps.

Reply:
Yes that helps, your changes in _updateProjection are simpler to follow.